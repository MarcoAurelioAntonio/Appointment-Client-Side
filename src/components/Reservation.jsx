import React from 'react';
import PropTypes from 'prop-types';

const Reservation = ({ reservation }) => (
  <div className="card">
    <img className="card-img" src="/images/course-1.jpg" alt="Avatar" />
    <div className="container">
      <p className="course-name">{reservation.course.name}</p>
      <div className="course-info">
        <p>Start date :</p>
        <span>{reservation.course.start_date}</span>
      </div>
      <div className="course-info">
        <p>End date :</p>
        <span>{reservation.course.end_date}</span>
      </div>
      <div className="card-price">
        <p>Price :</p>
        <span>{`$ ${reservation.course.price}`}</span>
      </div>
    </div>
  </div>
);

export default Reservation;

Reservation.propTypes = {
  reservation: PropTypes.shape().isRequired,
};
