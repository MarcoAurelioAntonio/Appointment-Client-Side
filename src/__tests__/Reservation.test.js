import { render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Reservation from '../components/Reservation';

describe('Tests for <Reservation /> component', () => {
  const reservationItem = {
    id: 1,
    user_id: 1,
    course_id: 1,
    created_at: '2023-06-20T22:04:22.218Z',
    updated_at: '2023-06-20T22:04:22.218Z',
    course: {
      id: 1,
      name: 'Master Data Structures',
      description: 'Description related to the course',
      course_type: 'Programming',
      price: '200.0',
      start_date: '2023-06-21',
      end_date: '2023-07-21',
      created_at: '2023-06-20T22:03:45.571Z',
      updated_at: '2023-06-20T22:03:45.861Z',
      image_url: 'http://localhost:3000/rails/active_storage/blobs/redirect/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBCZz09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--2e2c6f24621c425f569dcfdc5f64d41d7aa1f592/pexels-karolina-grabowska-4497759.jpg',
    },
  };

  it('The name of the course should be display properly.', () => {
    render(
      <BrowserRouter>
        <Reservation reservation={reservationItem} />
        ,
      </BrowserRouter>,
    );
    const element = screen.getByText(reservationItem.course.name);
    expect(element.textContent).toBe(reservationItem.course.name);
  });

  it('Should match the snapshot', () => {
    expect(renderer.create(
      <BrowserRouter><Reservation reservation={reservationItem} /></BrowserRouter>,
    ).toJSON()).toMatchSnapshot();
  });
});
