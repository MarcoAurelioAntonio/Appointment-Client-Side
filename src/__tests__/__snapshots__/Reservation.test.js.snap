// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Reservation /> component Should match the snapshot 1`] = `
<div
  className="card"
>
  <img
    alt="Avatar"
    className="card-img"
    src="http://localhost:3000/rails/active_storage/blobs/redirect/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBCZz09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--2e2c6f24621c425f569dcfdc5f64d41d7aa1f592/pexels-karolina-grabowska-4497759.jpg"
  />
  <div
    className="container"
  >
    <p
      className="course-name"
    >
      Master Data Structures
    </p>
    <div
      className="course-info"
    >
      <p>
        Start date :
      </p>
      <span>
        2023-06-21
      </span>
    </div>
    <div
      className="course-info"
    >
      <p>
        End date :
      </p>
      <span>
        2023-07-21
      </span>
    </div>
    <div
      className="card-price"
    >
      <p>
        Price :
      </p>
      <span>
        $ 200.0
      </span>
    </div>
  </div>
</div>
`;
